internal/common/utils.go

```diff
@@ -243,3 +239,50 @@ func GenerateDeleteOptions() *client.DeleteOptions {
 	}
 	return &delOpt
 }
+
+// RetryRoundTripper include anything here that's useful during middleware processing
+type RetryRoundTripper struct {
+	transport http.RoundTripper
+	backoff   wait.Backoff
+	log       logr.Logger
+}
+
+// RetryMiddleware pass this into your client to make it retriable with configured backoff
+func RetryMiddleware(logger logr.Logger) func(rt http.RoundTripper) http.RoundTripper {
```

Comment:
Maybe RetryTransportMiddleware is more clear?

Reply:
Hmm intentionally didnt add transport as it's actually a WrapperFunc that gets returned and used by Wrap()...

How about `RetryWrapMiddleware` instead?